### PR TITLE
[#136342237] Move suggestions

### PR DESCRIPTION
**Pull request by ArthurZaharov**

### CODE REVIEW

File: app/controllers/suggestions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class SuggestionsController < ApplicationController
+  expose_decorated :suggestions, -> { Suggestion.includes(:user).sorted }
+  expose_decorated :suggestion
+  expose_decorated :comments, -> { suggestion.comments.includes(:user) }
+  expose_decorated :comment, -> { suggestion.comments.new }
```
Comment (vladimirbazhanov) [inline app/controllers/suggestions_controller.rb]: there is no need to decorate new empty object I think

File: app/decorators/comment_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class CommentDecorator < ApplicationDecorator
+  delegate :text
+
+  def created_ago
```
Comment (vladimirbazhanov) [inline app/decorators/comment_decorator.rb]: this can be moved to ApplicationDecorator I think

File: app/decorators/comment_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    "#{h.time_ago_in_words(object.created_at)} ago"
+  end
+
+  def user_full_name
+    return user.full_name if user.hr?
```
Comment (vladimirbazhanov) [inline app/decorators/comment_decorator.rb]: So HR will see full name in any case?

File: app/decorators/comment_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    return "Anonymously" if comment.suggestion.hide_username
+
+    user.full_name
+  end
+
+  def user
+    @user ||= object.user.decorate
```
Comment (vladimirbazhanov) [inline app/decorators/comment_decorator.rb]: it's better to decorates_association :user if we want full object be visible

File: app/decorators/suggestion_decorator.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+class SuggestionDecorator < ApplicationDecorator
+  delegate :title, :description, :hide_username
+  decorates_association :comments
+
+  def created_ago
+    "#{h.time_ago_in_words(object.created_at)} тому назад"
```
Comment (vladimirbazhanov) [inline app/decorators/suggestion_decorator.rb]: to ApplicationDecorator too?

File: app/decorators/suggestion_decorator.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  def user_full_name
+    object.hide_username ? "Анонимус" : user.full_name
+  end
+
+  def user
+    @user ||= object.user.decorate
```
Comment (vladimirbazhanov) [inline app/decorators/suggestion_decorator.rb]: as in previous comment, decorate association?

File: app/decorators/comment_decorator.rb (hunk 3 above)
Comment (ArthurZaharov) [inline app/decorators/comment_decorator.rb]: No. User will see HR username anyway. I agree that there is strange logic.

**Pull request merged**